src/os/FileStore.cc

```diff
@@ -3355,59 +3355,30 @@ int FileStore::_do_sparse_copy_range(int from, int to, uint64_t srcoff, uint64_t
 {
   dout(20) << __func__ << " " << srcoff << "~" << len << " to " << dstoff << dendl;
   int r = 0;
-  struct fiemap *fiemap = NULL;
-
+  map<uint64_t, uint64_t> exomap;
+  int64_t written = 0;
```

Comment:
would be ideal if this definition (the variable of "written") could be move close to where it is first used, i.e. https://github.com/ceph/ceph/pull/5312/files#diff-0d2d3deefdf5127945a40ed3e04c154fR3372 .